readthedocs/api/v3/views.py

```diff
@@ -589,6 +589,10 @@ class UsersViewSet(
     serializer_class = UserSerializer
     queryset = User.objects.none()
     permission_classes = (IsAuthenticated,)
+    # We are using the username as the lookup field,
+    # by default, DRF does not allow dots and `/`,
+    # but we allow usernames to have dots.
+    lookup_value_regex = "[^/]+"
```

Comment:
Shouldn't we use something more restrictive here? Like `[0-9a-fA-Z@.+-_]`

I found it on https://github.com/django/django/blob/main/django/contrib/auth/models.py#L350C69-L350C78

Reply:
Actually, I found the regex itself: https://github.com/django/django/blob/main/django/contrib/auth/validators.py#L20

```
r"^[\w.@+-]+\Z"
```

We should use this one.

Reply:
No strong opinion. This is just to allow capturing the username, validation is done when querying for that username, there aren't any security implications.

Reply:
I think what we want is to re-use that regex and don't maintain ours.

```suggestion
    lookup_value_regex = UnicodeUsernameValidator.regex
```

Reply:
But we aren't using that regex for validation, just for routing in this case.

Reply:
What's the difference? 

The regex is the same, right?

Reply:
The problem here is that the URL doesn't allow dots, so the request can't be routed to the correct view, that view makes sure the username is valid. That problem is solved with the code from this PR. Exporting or writing a custom regex is just adding extra complexity where we don't need it.